app/controllers/doorkeeper/authorizations_controller.rb

```diff
@@ -1,8 +1,15 @@
 module Doorkeeper
   class AuthorizationsController < Doorkeeper::ApplicationController
-    before_action :authenticate_resource_owner!
+    before_action do
+      render_prompt_error(:login_required) unless begin
```

Comment:
Let's extract this block into some method.

Also, what if some legacy projects will inherit from this controller - do they still get backward compatibility ?